zipline/finance/performance/period.py

```diff
@@ -203,9 +200,12 @@ def calculate_performance(self):
 
     def record_order(self, order):
         if self.keep_orders:
-            dt_orders = self.orders_by_modified[order.dt]
-            if order.id in dt_orders:
-                del dt_orders[order.id]
+            try:
+                dt_orders = self.orders_by_modified[order.dt]
+                if order.id in dt_orders:
+                    del dt_orders[order.id]
```

Comment:
Also, right below the `del` is used to preserve the FIFO ordering of orders, but that is not relevant with `orders_by_modified` which is just a normal dictionary... but that probably explains why the `del` crept in there.